File: 2024/02A.py
import sys

def is_safe(report):
    diffs = set([report[i] - report[i - 1] for i in range(1,len(report))])
    if diffs.issubset({1,2,3}) or diffs.issubset({-1,-2,-3}):
        return True
    return False

def solve(input):
    reports = [[int(word) for word in line.split()] for line in input.splitlines()]   
    return sum(is_safe(report) for report in reports)
     
if __name__ == '__main__':
    print(solve(open(sys.argv[1]).read())) 

File: 2024/02A.js

function isSafe(report) {
  const diffs = new Set();
  for (let i = 1 ; i < report.length ; i++) {
    diffs.add(report[i-1] - report[i]);
  }
  const max = Math.max(...diffs);
  const min = Math.min(...diffs);
  if ((min > 0 && max <=3) || (min >= -3 && max < 0)) {
    return true;
  }
  return false;
}
function solve(input) {
  const reports = input.split('\n').map((row) => row.split(' ').map((v) => parseInt(v))).sort();
  return reports.filter(isSafe).length;
}

console.log(solve(require('node:fs').readFileSync(process.argv[2], 'utf8').trim()));